### PR TITLE
Prac 06 feedback

### PR DESCRIPTION
**Pull request by ShaneLind**
Hi @JackKerlin, please have a look at my code and let me know how I can improve. Thanks.

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+        self._odometer += distance
+        return distance
+
+    def __str__(self):
+        text = str(f"{self.name}, fuel={self.fuel}, odometer={self._odometer}")
```
Comment (JackKerlin) [inline prac_06/car.py]: The str() is unnecessary, you can just format as text = f"...

File: prac_06/guitars.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+CP1404 Practical - Client code to use the Guitar class.
+"""
+from guitar import Guitar
+
+my_guitars = []
+
+print("My Guitars!")
+name = input("Name: ")
+while not name == "":
+    year = input("Year: ")
```
Comment (JackKerlin) [inline prac_06/guitars.py]: An int() is needed as else is_vintage() will subtract a string from a integer, giving an error

File: prac_06/guitars.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    cost = input("Cost: ")
+    my_guitars += [name, year, cost]
```
Comment (JackKerlin) [inline prac_06/guitars.py]: this adds three non-Guitar class objects to the list, I suggest replacing with adding [Guitar(name, year, cost)] to add a single Guitar object

File: prac_06/guitars.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+"""
+from guitar import Guitar
+
+my_guitars = []
+
+print("My Guitars!")
+name = input("Name: ")
+while not name == "":
+    year = input("Year: ")
+    cost = input("Cost: ")
```
Comment (JackKerlin) [inline prac_06/guitars.py]: A float() is needed as else the f string formatting for decimals will give an error

File: prac_06/languages.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+# print(python)
+# print(ruby)
+# print(visual_basic)
+
+programming_languages = [python, ruby, visual_basic]
+print("The dynamically typed languages are:")
+for language in programming_languages:
+    if ProgrammingLanguage.is_dynamic(language):
```
Comment (JackKerlin) [inline prac_06/languages.py:19]: if you want you can format the method as language.is_dynamic() without the class's name, though it is good either way

**Review by JackKerlin — COMMENTED** [review]
Good code overall, well formatted, just some minor issues with the code in the while loop in guitars.py

File: prac_06/guitars.py (hunk 3 above)
Comment (ShaneLind) [reply prac_06/guitars.py]: That makes a lot more sense, thx

File: prac_06/guitars.py (hunk 4 above)
Comment (ShaneLind) [reply prac_06/guitars.py]: I never checked decimals, thanks for that

File: prac_06/languages.py (hunk 5 above)
Comment (lindsaymarkward) [reply prac_06/languages.py:19]: Jack's suggestion is good - please use the normal concise syntax, not this unbound version.

**Pull request merged**